packages/core/src/repository/cache.repository.ts

```diff
@@ -54,7 +56,28 @@ export class CacheRepository<T> implements GetRepository<T>, PutRepository<T>, D
             case MainOperation:
                 return this.getMain.get(query);
             case CacheOperation:
-                return this.getCache.get(query);
+                return this.getCache.get(query).then((value: T) => {
+                    if (value == null) {
+                        throw new NotFoundError();
+                    }
+                    if (!this.validator.isObjectValid(value)) {
+                        throw new NotValidError();
+                    }
+                    return value;
+                }).catch((err: Error) => {
+                    if (err instanceof NotValidError || err instanceof NotFoundError) {
+                        return this.get(query, new MainOperation()).catch((finalError: Error) => {
```

Comment:
Take my comment with caution as I am not aware of the inner working on Harmony TS or I might be not understanding the code.
I see something strange in here, you are inside a CacheOperation but in the error you retrying with MainOperation. This is the expected behavior for CacheSyncOperaion but not for CacheOperation.
What I've implemented on harmony-kotlin is the following:

- Try getting from cache
- If any error happen pass it to the fallback lambda to see if I can retry
- If I can retry just try with cache again ignoring any validation

Reply:
@franmontiel you are right. But this was fixed earlier. The code you're seeing is outdated or you're reviewing the first commit only.

Reply:
My apologies I didn't notice, it had it opened since this morning pending to write the comment.